bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -276,20 +274,93 @@ private void manageCurPosIfEnvContainsTarget(final SPTarget target, final Runnab
             t.addWatcher(posWatcher);
             refreshAll();
         });
+        return true;
+    }
+
+    /**
+     * Determine whether or not the auto group has changed for a specific observation.
+     */
+    private boolean autoGroupChanged(final ISPObsComponent oldObsComp, final TargetObsComp oldTOC) {
+        final TargetObsComp newTOC = getDataObject();
+
+        // If either of the TargetObsComps are null, this was not triggered by an auto group change.
+        if (oldTOC == null || newTOC == null) {
+            return false;
+        }
+
+        // If either of the node keys are null, or they are defined but different, this is not an auto group change.
+        final SPNodeKey oldNodeKey = ImOption.apply(oldObsComp).map(ISPObsComponent::getNodeKey).getOrNull();
+        final SPNodeKey newNodeKey = ImOption.apply(getNode()).map(ISPObsComponent::getNodeKey).getOrNull();
+        if (oldNodeKey == null || newNodeKey == null) {
+            return false;
+        }
+        if (!oldNodeKey.equals(newNodeKey)) {
+            return false;
+        }
+
+        // Extract the target environments and check if the auto groups have changed.
+        final TargetEnvironment oldEnv    = oldTOC.getTargetEnvironment();
+        final TargetEnvironment newEnv    = newTOC.getTargetEnvironment();
+        final Option<GuideGroup> oldGpOpt = oldEnv.getGroups().headOption().filter(GuideGroup::isAutomatic);
+        final Option<GuideGroup> newGpOpt = newEnv.getGroups().headOption().filter(GuideGroup::isAutomatic);
+
+        // We only want to return false if the two groups exist and contain the same targets.
+        return oldGpOpt.forall(oldGp ->
+                newGpOpt.forall(newGp -> {
+                    // Extract the target names.
+                    final List<String> oldNames = oldGp.getTargets().map(SPTarget::getName).toList();
+                    final List<String> newNames = newGp.getTargets().map(SPTarget::getName).toList();
+                    return !(oldNames.size() == newNames.size() && oldNames.containsAll(newNames));
```

Comment:
I'm thinking that maybe it makes sense to add a `toMap` and `toJavaMap` to `GuideGroup` to compare, or even more, a method directly in `GuideGroup` that compares the group to another.

I don't think it's as simple as doing an equals op and that we will actually have to do some extraction since `==>>` is a sealed abstract class.